mono/utils/mono-mmap.c

```diff
@@ -287,10 +287,21 @@ mono_valloc (void *addr, size_t length, int flags, MonoMemAccountType type)
 #endif
 
 #if defined(__APPLE__) && defined(MAP_JIT)
```

Comment:
`#ifdef MAP_JIT` here precludes the ability to build on older SDK and run on newer OS (with hardened runtime).
It'd be good to `#define` it ourselves to allow for that. Copy the SDK header content. It is a little gross but the benefits are good and it is not unusual practice.